doc/api-design.yml

```diff
@@ -0,0 +1,903 @@
+openapi: 3.0.3
+info:
+  title: Quizzard API
+  description: API documentation for the Quizzard application.
+  version: 1.0.0
+servers:
+  - url: http://localhost:8080/api
+    description: Development server
+components:
+  securitySchemes:
+    bearerAuth:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: integer
+        username:
+          type: string
+        name:
+          type: string
+        email:
+          type: string
+          format: email
+        profile_picture:
+          type: string
+          format: uri
+        english_proficiency:
+          type: string
+          enum: [a1,a2,b1,b2,c1,c2]
+        points:
+          type: integer
+        created_quizzes:
+          type: integer
+        created_at:
+          type: string
+          format: date-time
+        updated_at:
+          type: string
+          format: date-time
+      required:
+        - username
+        - name
+        - email
+        - password
+        - english_proficiency
+    RegisterRequest:
+      type: object
+      properties:
+        username:
+          type: string
+        name:
+          type: string
+        email:
+          type: string
+          format: email
+        password:
+          type: string
+        english_proficiency:
+          type: string
+          enum: [a1,a2,b1,b2,c1,c2]
+      required:
+        - username
+        - name
+        - email
+        - password
+        - english_proficiency
+    LoginRequest:
+      type: object
+      properties:
+        username_or_email:
+          type: string
+        password:
+          type: string
+      required:
+        - username_or_email
+        - password
+    Quiz:
```

Comment:
Can we add a description field to quizzes as well?

Reply:
it is added